src/components/crowdsale/utils.js

```diff
@@ -274,14 +274,65 @@ let fillCrowdsalePageStoreDates = (startsAtMilliseconds, endsAtMilliseconds) =>
   logger.log('endDate:', endsAtMilliseconds)
 }
 
+/**
+ * Check if a crowdsale is finalized
+ * @param methods
+ * @param crowdsaleExecID
+ * @returns {Promise<*>}
+ */
 export const isFinalized = async ({ methods }, crowdsaleExecID) => {
   const { addr } = contractStore.abstractStorage
   let params = []
   if (crowdsaleExecID) {
     params.push(addr, crowdsaleExecID)
   }
-  const { is_finalized } = await methods.getCrowdsaleInfo(...params).call()
-  return is_finalized
+  const crowdsale = await methods.getCrowdsaleInfo(...params).call()
+  const { isMintedCappedCrowdsale, isDutchAuction } = crowdsaleStore
+  let isFinalized = false
+  if (isMintedCappedCrowdsale) {
+    //Value is finalized is the index #3
+    isFinalized = crowdsale[3]
+  } else if (isDutchAuction) {
+    //Value is finalized is the index #4
+    isFinalized = crowdsale[4]
+  }
+  logger.log(`Crowdsale Info: Is finalized`, isFinalized)
+  return isFinalized
+}
+
+/**
+ * Check if a crowdsale is ended
+ * @param methods
+ * @param crowdsaleExecID
+ * @returns {Promise<boolean>}
+ */
+export const isEnded = async ({ methods }, crowdsaleExecID) => {
+  const { addr } = contractStore.abstractStorage
+  let params = []
+  if (crowdsaleExecID) {
+    params.push(addr, crowdsaleExecID)
+  }
+  const crowdsaleStartAndEndTimes = await methods.getCrowdsaleStartAndEndTimes(...params).call()
+  logger.log(`Crowdsale start time:`, crowdsaleStartAndEndTimes[0])
+  logger.log(`Crowdsale end time::`, crowdsaleStartAndEndTimes[1])
+  const end_time = crowdsaleStartAndEndTimes[1]
+  return end_time * 1000 <= Date.now()
+}
+
+/**
+ * Check if a crowdsale is sold out
+ * @param methods
+ * @param crowdsaleExecID
+ * @returns {Promise<*>}
+ */
+export const isSoldOut = async ({ methods }, crowdsaleExecID) => {
```

Comment:
Just to remind for future PR, that in frames of this comment https://github.com/poanetwork/token-wizard/issues/1026#issuecomment-403885893 this function should be redefined

Reply:
Yes, we could leave `isSoldOut` function as is and create an additional, for example, `tokensRemaining` function, that will check remaining tokens at the current tier. And add this new function there, where we check the possibility to finalize crowdsale.